### PR TITLE
Feat/virtual joystick

### PR DESCRIPTION
**Pull request by JWS-aAI**
**New feature added: Basic Virtual Joystick**

Hi, I decide implement this feature because it appears to be in high demand. Hope you find interesting.

**Description:**
- Joystick can be activated by button on bottom (see picture)
- For resolutions less than 650px wide it's active by default
- When resolution ir less than 650px and chat is active, joystick is hidden because it interferes with the chat

Captura de Pantalla 2023-02-20 a las 1 20 20

For implement this, I use this component: https://github.com/elmarti/react-joystick-component

Note: This feature will conflict with this PR:  #.62
I can take care of resolving conflicts

Related issues: 
- #.23
- #.60

### CODE REVIEW

File: client/src/stores/UserStore.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ export const userSlice = createSlice({
     videoConnected: false,
     loggedIn: false,
     playerNameMap: new Map<string, string>(),
+    showJoystick: window.innerWidth < 650,
```
Comment (PR-Review) [inline client/src/stores/UserStore.ts:21]: The use of `window.innerWidth` directly in the Redux slice initial state may cause issues during server-side rendering or testing, as `window` is not defined outside the browser context. Consider initializing this state in a component or a setup function that is guaranteed to run in the browser.

**Comment by PR-Review** [comment]
Thank you for your contribution to the SkyOffice project. I have finished reviewing the pull request for the virtual joystick feature. Overall, the implementation seems solid, and the feature is a valuable addition to the application. I've left some comments regarding best practices and maintainability, which I hope you'll find helpful.

Please address the comments I've made, especially regarding the use of `window.innerWidth` in the Redux slice and the potential need for TypeScript conversion for consistency. Once these points are addressed, we should be closer to merging this feature into the main branch.

If you have any questions or need further clarification on any of the points raised, feel free to reach out. Looking forward to seeing this feature in action!

**Comment by JWS-aAI** [comment]
@PR-Review just shortly, what should i do first?

**Comment by PR-Review** [comment]
@JWS-aAI Firstly, address the comment regarding the direct use of `window.innerWidth` in the Redux slice initial state. This could cause issues during server-side rendering or testing. You should initialize this state in a component or a setup function that runs in the browser context.

Secondly, consider converting the `Joystick.jsx` file to TypeScript to maintain consistency with the rest of the project's codebase.

Lastly, review the use of the magic number `650` for screen width and either document it or replace it with a named constant for better code readability and maintainability.

Once these changes are made, please update the pull request, and we can proceed with the next steps.

File: client/src/stores/UserStore.ts (hunk 1 above)
Comment (JWS-aAI) [reply client/src/stores/UserStore.ts:21]: @PR-Review ok this wont be an issue.

Comment (PR-Review) [reply client/src/stores/UserStore.ts:21]: @JWS-aAI Understood. If you're confident that the use of `window.innerWidth` won't cause any issues in your specific setup, you can proceed with the other points raised in the review. Just ensure that any potential edge cases are handled, such as when the code might execute in a non-browser environment.

Please proceed with the conversion of `Joystick.jsx` to TypeScript for consistency and address the use of the magic number `650`. Once these updates are made, the pull request can move forward. If you encounter any issues or have further questions, feel free to reach out.

**Pull request closed without merging**